### PR TITLE
Feature/10 show only selected files

### PR DESCRIPTION
**Pull request by MicSm**

### CODE REVIEW

File: src/extension.js [hunk 1]
```diff
@@ -270,7 +270,31 @@ function searchTags()
 
 	//Case 2. Many tags found
 	} else if (displayFiles.length > 0) {
-		vscode.window.showQuickPick(displayFiles, {matchOnDescription: true, matchOnDetail: true})
+
+		// extract available extensions from ctagssh settings if any
+		let conf = vscode.workspace.getConfiguration('ctagssh');
+		let CTagSSH_showExtensions = [];
```
Comment (Kandimus) [inline src/extension.js:276]: **CTagSSH_** prefix used only for global variables

File: src/extension.js [hunk 2]
```diff
@@ -270,7 +270,31 @@ function searchTags()
...
+		let filteredFiles = [];
+
+		if ("" !== conf.showExtensions) {
+		
+			CTagSSH_showExtensions = conf.showExtensions.split(/[,;\s]+/).filter((/** @type {string} */ element) => element !== "").map((/** @type {string} */ element) => element.toLowerCase());
+		}
+		// if there are available fltering extensions then filter files list
+		if (0 !== CTagSSH_showExtensions.length) {
+
+			const path = require('path');
```
Comment (Kandimus) [inline src/extension.js:286]: const path present in line 2

File: src/extension.js [hunk 3]
```diff
@@ -270,7 +270,31 @@ function searchTags()
...
+			filteredFiles = displayFiles.filter((/** @type {{ filePath: string; }} */ element) => CTagSSH_showExtensions.includes(path.extname(element.filePath).toLowerCase().substring(1)));
+		}
+		if (0 === filteredFiles.length) {
+			filteredFiles = displayFiles;
+		}
+		// enumeration of list of matching tags
+		for (let i = 0; i < filteredFiles.length; i++) {
```
Comment (Kandimus) [inline src/extension.js:293]: may be to use ForEach?

File: src/extension.js [hunk 4]
```diff
@@ -133,6 +132,7 @@ function activate(context)
 	context.subscriptions.push(vscode.commands.registerCommand('ctagssh.test', async () => {
 		var a = 1;
 	}));
+
```
Comment (Kandimus) [inline src/extension.js:135]: empty line. why? )))

**Review by Kandimus — CHANGES_REQUESTED** [review]

File: src/extension.js (hunk 1 above)
Comment (MicSm) [reply src/extension.js:276]: done

File: src/extension.js (hunk 4 above)
Comment (MicSm) [reply src/extension.js:135]: done

File: src/extension.js (hunk 2 above)
Comment (MicSm) [reply src/extension.js:286]: done

File: src/extension.js (hunk 3 above)
Comment (MicSm) [reply src/extension.js]: done

**Review by Kandimus — APPROVED** [review]

**Pull request merged**